fsspec/spec.py

```diff
@@ -381,7 +405,7 @@ def _ls_from_cache(self, path):
         except KeyError:
             pass
 
-    def walk(self, path, maxdepth=None, topdown=True, on_error="omit", **kwargs):
+    def walk(self, path: str, maxdepth=None, topdown=True, on_error="omit", **kwargs):
```

Comment:
nit: on_error has only a few valid values, right? We should make it a Union of Type Literals